### PR TITLE
wrappers: run update-desktop-database after add/remove of desktop files

### PR DESCRIPTION
**Pull request by mvo5**
Closes: LP:#1597417

### CODE REVIEW

File: wrappers/desktop.go [hunk 1]
```diff
@@ -163,6 +164,15 @@ func sanitizeDesktopFile(s *snap.Info, rawcontent []byte) []byte {
 	return []byte(strings.Join(newContent, "\n"))
 }
 
+func updateDesktopDatabase() error {
+	if _, err := exec.LookPath("update-desktop-database"); err == nil {
```
Comment (zyga) [inline wrappers/desktop.go]: I'd add a `// TODO: switch on release.ReleaseInfo.ID to decide what distro-specific hook to run`

Comment (mvo5) [inline wrappers/desktop.go]: Hm, my understanding is that update-desktop-database is a xdg thing, so we probably want to run it everywhere where it is available? Happy to run more and distro specific stuff once we have a use-case.

File: wrappers/desktop.go [hunk 2]
```diff
@@ -163,6 +164,15 @@ func sanitizeDesktopFile(s *snap.Info, rawcontent []byte) []byte {
...
+		if output, err := exec.Command("update-desktop-database").CombinedOutput(); err != nil {
+			return fmt.Errorf("cannot update-desktop-database %q: %s", output, err)
+		}
+	}
```
Comment (chipaca) [inline wrappers/desktop.go]: maybe log something?

File: wrappers/desktop.go [hunk 3]
```diff
@@ -204,5 +221,12 @@ func RemoveSnapDesktopFiles(s *snap.Info) error {
 		os.Remove(f)
 	}
 
+	// updates mime info etc
+	if len(activeDesktopFiles) > 0 {
```
Comment (chipaca) [inline wrappers/desktop.go]: maybe move the `if len()` into the `update` itself?

**Comment by chipaca** [comment]
Maybe :+1: ?

**Pull request merged**